### PR TITLE
Add custom error handler to login

### LINKED ISSUE #150: Invalid login gives us invalid response
**Describe the bug**

Any invalid combination of email / password returns 500 error. **To Reproduce**
1. `POST /api/auth/login` with any invalid information

**Expected behavior**

Should return 401 Unauthenticated, or something else.

### PR DESCRIPTION
**Pull request by YiFeiZhang2**
# Description

Added a middleware wrapper around passport to add a custom error handler in case of invalid login, and errors while logging in. Invalid account/password now gives an error 401 with respective auth 401 error.

Fixes #150

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Added a test to auth.test for the case where the password does not match the email.

### CODE REVIEW

File: middlewares/auth.middleware.js [hunk 1]
```diff
@@ -20,6 +21,36 @@ const Constants = {
...
+                });
+            }
+            if (!user) {
+                return next({
+                    status: 401,
+                    message: Constants.Error.AUTH_401_MESSAGE,
+                    error: {}
+                });
+            }
+            req.login(user, loginErr => {
```
Comment (pierreTklein) [inline middlewares/auth.middleware.js]: Rly small -- Can you put parentheses around `loginErr`

Comment (pierreTklein) [reply middlewares/auth.middleware.js]: Or, make this a non-arrow function

File: services/auth.service.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = {
                 if (!!account) {
                     done(null, account);
                 } else {
-                    done("Invalid email or password.", false);
+                    done(null, false);
```
Comment (pierreTklein) [inline services/auth.service.js]: What happens if we leave this message in?

Comment (YiFeiZhang2) [reply services/auth.service.js]: If we use the default error handler, we get an error 500, as the first argument is error. For our custom error handler, I've followed the same format, so we will also get a 500 error.

Comment (pierreTklein) [reply services/auth.service.js]: Oh I see, so by putting in `Invalid email or password`, it doesn't call your custom callback?

**Pull request merged**